src/core/image.js

```diff
@@ -86,11 +86,9 @@ var PDFImage = (function PDFImageClosure() {
     if (dict.has('Filter')) {
       var filter = dict.get('Filter').name;
       if (filter === 'JPXDecode') {
-        var jpxImage = new JpxImage();
-        jpxImage.parseImageProperties(image.stream);
-        image.stream.reset();
-        image.bitsPerComponent = jpxImage.bitsPerComponent;
-        image.numComps = jpxImage.componentsCount;
+        const jpxProperties = JpxImage.parseImageProperties(image.stream);
```

Comment:
It seems that `JpxImage` is always used as the temporary object to store output (your `jpxProperties` here).

https://github.com/mozilla/pdf.js/blob/ae07adf143a7680480fdb34df63910466448f79b/src/core/jpx_stream.js#L53-L54
https://github.com/mozilla/pdf.js/blob/620da6f4df6eb883ff89580b643ef0516035e0d0/src/core/image.js#L89-L90

Since `parseImageProperties` does not depend on pre-existing `JpxImage` instance methods or members, it seems to make sense to turn it into a static method. (in the past this was different: 894d9fe085b369b8857b87e277bbf67bc98fa2f6).

However, I'd still like to see this commit removed, for consistency with our other image decoding implementations (jpg.js, jbig2.js) also use the pattern of a instantiating a temporary class for parsing images.

Reply:
Fair point regarding the consistency, the re-factoring commit has been removed.